src/loki/search.cc

```diff
@@ -448,16 +439,17 @@ struct bin_handler_t {
     // iterate over the edges in the bin
     auto tile = begin->cur_tile;
     auto edges = tile->GetBin(begin->bin_index);
-    for (auto e : edges) {
+    for (auto edge_id : edges) {
       // get the tile and edge
-      if (!reader.GetGraphTile(e, tile)) {
+      if (!reader.GetGraphTile(edge_id, tile)) {
         continue;
       }
 
       // no thanks on this one or its evil twin
-      const auto* edge = tile->directededge(e);
-      if (edge_filter(edge) == 0.0f && (!(e = reader.GetOpposingEdgeId(e, tile)).Is_Valid() ||
-                                        edge_filter(edge = tile->directededge(e)) == 0.0f)) {
+      const auto* edge = tile->directededge(edge_id);
+      if (edge_filter(edge) == 0.0f &&
+          (!(edge_id = reader.GetOpposingEdgeId(edge_id, tile)).Is_Valid() ||
+           edge_filter(edge = tile->directededge(edge_id)) == 0.0f)) {
```

Comment:
Any way you could re-write this set of branches to be clearer?  The `edge_id =` and `edge =` re-assignments within the conditional are *really* easy to miss.  It's not obvious that `edge` and `edge_id` may be re-assigned after this statement.

How about:

```cpp
// If this edge is filtered out, try switching edge_id and edge to the opposing direction.
// If those are also filtered out, we're done with this edge.
if (edge_filter(edge) == 0.0f) {
  // switch to the reverse edge
  edge_id = reader.GetOpposingEdgeId(edge_id, tile);
  if (edge_id.Is_Valid()) {
    edge = tile->directededge(edge_id);
    if (edge_filter(edge) == 0.0f) {
      continue;
    }
  } else {
    continue;
  }
}
```